### PR TITLE
[merged] trust show GPG key ID

### LINKED ISSUE #661: atomic trust show --json should output parsed policy
Currently `atomic trust show --json` just prints the policy.json file. More useful (and expected) is to output the same data in json format. We would then use a --policy-file flag to display the raw policy config. So this...

```
$ sudo atomic trust show
docker.io                           accept
docker.io/aweiteka                  signed   https://s3.amazonaws.com/aweiteka-sigstore
* (default)                         reject
```

Would output something like the policy file but with sigstore URI information:

```
{
	"default": [{
		"type": "reject",
                "sigstore-staging": "file:///var/lib/atomic/sigstore"
	}],
	"transports": {
		"docker": {
			"docker.io/aweiteka": [{
				"keyPath": "/etc/pki/containers/aweiteka@gmail.com",
				"keyType": "GPGKeys",
				"type": "signedBy",
				"sigstore": "https://s3.amazonaws.com/aweiteka-sigstore"
			}],
			"docker.io": [{
				"type": "insecureAcceptAnything"
			}]
		},
		"atomic": {}
	}
}
```

### PR DESCRIPTION
**Pull request by aweiteka**
Show signed GPG key IDs on output
Option to output raw policy file
Option to output full trust as json for machine parsing
Add key parsing to tests using actual public keys

resolves #661 and #646 and #703 

```
$ sudo atomic trust show
* (default)                         accept
docker.io                           reject
docker.io/aweiteka                  signed                               https://s3.aws.com/aweiteka-sigstore/
docker.io/billy                     signed Billy Bob                     https://s3.aws.com/aweiteka-sigstore/
registry.access.redhat.com          signed aweiteka@redhat.com           https://example.com/sigstore
```

### CODE REVIEW

File: Atomic/trust.py [hunk 1]
```diff
@@ -351,26 +352,73 @@ def show(self):
...
+            else:
+                cmd = ["gpg2", "--list-packets", key]
+            try:
+                results = util.check_output(cmd)
+            except util.FileNotFound:
+                results = ""
+            lines = results.split(b'\n')
+            for line in lines:
+                line = str(line)
+                if "user ID packet" in line:
```
Comment (rhatdan) [inline Atomic/trust.py]: This seems a little fragile.  What happens if I use a different language?

**Comment by rhatdan** [comment]
bash completions.

File: Atomic/trust.py [hunk 2]
```diff
@@ -351,26 +352,73 @@ def show(self):
...
                 if "transports" in policy:
                     for tt in TRANSPORT_TYPES:
                         if tt in policy["transports"]:
-                            for key, value in policy["transports"][tt].items():
-                                table[key] = { "type": value[0]["type"] }
+                            for key, values in policy["transports"][tt].items():
+                                table[key] = { "type": values[0]["type"] }
+                                table[key]["keys"] = []
+                                for v in values:
+                                    if "keyPath" in v:
```
Comment (rhatdan) [inline Atomic/trust.py]: Would you ever want keyPath and keyValue at the same time?

File: Atomic/util.py [hunk 3]
```diff
@@ -184,6 +184,8 @@ def check_output(cmd, env=None, stdin=None, stderr=None):
         if e.args[0] == errno.ENOENT:
             raise FileNotFound("Cannot find file: `{}`".format(cmd[0]))
         raise
+    except subprocess.CalledProcessError:
+        return False
```
Comment (aweiteka) [inline Atomic/util.py]: Need to catch this in the caller

File: Atomic/trust.py (hunk 1 above)
Comment (aweiteka) [inline Atomic/trust.py]: After some [research](http://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=blob_plain;f=doc/DETAILS) it looks like the intended machine parsing output is to use `--with-colons` for a stable, colon-delimited output.

File: Atomic/trust.py (hunk 2 above)
Comment (rhatdan) [inline Atomic/trust.py]: Not with the elif
```
+                                    if "keyPath" in v:
+                                        table[key]["keys"].append(v["keyPath"])
+                                    elif "keyData" in v:
+                                        table[key]["keys"].append(v["keyData"])
```
Should this be?

```
+                                    if "keyPath" in v:
+                                        table[key]["keys"].append(v["keyPath"])
+                                    if "keyData" in v:
+                                        table[key]["keys"].append(v["keyData"])
```

Comment (aweiteka) [inline Atomic/trust.py]: Got it. Yes.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**